### PR TITLE
Refactor readme rendering logic

### PR DESCRIPTION
**Pull request by jonparrott**
Towards #3739
Original context: https://github.com/pypa/warehouse/pull/3745

### CODE REVIEW

**Comment by brainwane** [comment]
@aalmazan can you review this, if you have time?

**Comment by di** [comment]
I am (very slowly) reviewing & testing this.

File: tests/unit/packaging/test_views.py [hunk 1]
```diff
@@ -180,6 +180,7 @@ def test_detail_renders(self, db_request):
             "project": project,
             "release": releases[1],
             "files": [files[1]],
+            "description": None,
```
[the remark this answers is not in the document]
Comment (jonparrott) [reply tests/unit/packaging/test_views.py]: Done. :)

**Comment by jonparrott** [comment]
@di this should be ready to go now.

**Pull request merged**